main.go

```diff
@@ -80,23 +86,47 @@ func realMain(args []string) int {
 		return usageError("no JSON documents to validate")
 	}
 
+
 	// Compile target schema
 	sl := gojsonschema.NewSchemaLoader()
-	schemaUri := fileUri(*schemaFlag)
+	schemaUri := *schemaFlag
 	for _, ref := range refFlags {
 		for _, p := range glob(ref) {
 			uri := fileUri(p)
 			if uri == schemaUri {
 				continue
 			}
-			loader := gojsonschema.NewReferenceLoader(uri)
+			var loader gojsonschema.JSONLoader = nil
+
+			if strings.HasSuffix(uri, ".yaml") || strings.HasSuffix(uri, ".yml") {
+				valuesJSON, err := convertToJson(uri)
+				if err != nil {
+					log.Fatal("unable to parse YAML schema\n", err)
+				}
+				loader = gojsonschema.NewBytesLoader(valuesJSON)
+			} else {
+				loader = gojsonschema.NewReferenceLoader(uri)
+			}
```

Comment:
Can you wrap this in a function that returns the `gojsonschema.JSONLoader` since the yaml check is needed in a couple of places.

Reply:
Done. Pulled into the jsonloader func that you suggested below.